File: api/app/core/whatsapp.py
from urllib.parse import quote


def whatsapp_url(phone: str | None, text: str) -> str | None:
    if not phone:
        return None
    digits = "".join(ch for ch in phone if ch.isdigit())
    if not digits:
        return None
    return f"https://wa.me/{digits}?text={quote(text)}"


def profile_message(site_name: str) -> str:
    return f"Hola {site_name}, quiero hacer una consulta."


def publication_message(site_name: str, title: str) -> str:
    return f'Hola {site_name}, vi tu publicación "{title}" y quisiera consultar.'

File: web/src/lib/whatsapp.ts
export function whatsappHref(phone: string | null | undefined, text: string): string | null {
  if (!phone) return null;
  const digits = phone.replace(/\D/g, "");
  if (!digits) return null;
  return `https://wa.me/${digits}?text=${encodeURIComponent(text)}`;
}

export function profileWhatsappText(siteName: string): string {
  return `Hola ${siteName}, quiero hacer una consulta.`;
}

export function publicationWhatsappText(siteName: string, title: string): string {
  return `Hola ${siteName}, vi tu publicación "${title}" y quisiera consultar.`;
}
